code/game/objects/effects/mines.dm

```diff
@@ -7,19 +81,42 @@
 	icon_state = "uglymine"
 	/// We manually check to see if we've been triggered in case multiple atoms cross us in the time between the mine being triggered and it actually deleting, to avoid a race condition with multiple detonations
 	var/triggered = FALSE
+	var/smart_mine = FALSE
+	var/disarm_time = 20 SECONDS
+	var/disarm_product = /obj/item/deployablemine // The item dropped on disarm.
+
+/obj/effect/mine/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/multitool))
+		to_chat(user, "<span class='notice'>You begin to disarm the [src]...</span>")
+		if(do_after(user, disarm_time, target = src))
+			to_chat(user, "<span class='notice'>You disarm the [src].</span>")
+			new disarm_product(src.loc)
+			qdel(src)
+			return
+	return ..()
 
 /obj/effect/mine/proc/mineEffect(mob/victim)
 	to_chat(victim, "<span class='danger'>*click*</span>")
 
-/obj/effect/mine/Crossed(atom/movable/AM)
-	if(triggered || !isturf(loc))
+/obj/effect/mine/Crossed(atom/movable/AM as mob|obj)
+	if(!isturf(loc) || AM.throwing || (AM.movement_type & (FLYING | FLOATING)) || !AM.has_gravity())
 		return
 	. = ..()
 
-	if(AM.movement_type & FLYING)
-		return
-
-	triggermine(AM)
+	if(ismob(AM))
+		checksmartmine(AM)
+	else
+		triggermine(AM)
+
+/obj/effect/mine/proc/checksmartmine(mob/target)
+	if(target)
+		if(smart_mine) // Smart mine
+			if(!HAS_TRAIT(target, TRAIT_MINDSHIELD)) // Target doesn't have a mindshield
+				triggermine(target)
+			else if(istype(target.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat)) // Has mindshield AND tinfoil hat
```

Comment:
You probably took this from a different codebase, but mindshields and tinfoil hats are not something fitting our setting.

Reply:
Mindshields are still coded in that codebase and are printable from protolathes.

Reply:
Once researched   that is all these mines  aren't available  as i cant think of anything  that would come of it maybe it can be used fo  some High tech Admin event i want  to leave  them in here for awhile